roboRIO-2024/src/main/java/frc/robot/subsystems/Odometry.java

```diff
@@ -0,0 +1,132 @@
+package frc.robot.subsystems;
+
+import com.ctre.phoenix.sensors.PigeonIMU;
+import com.pathplanner.lib.auto.AutoBuilder;
+import com.pathplanner.lib.util.HolonomicPathFollowerConfig;
+import com.pathplanner.lib.util.PIDConstants;
+import com.pathplanner.lib.util.ReplanningConfig;
+import com.revrobotics.RelativeEncoder;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.kinematics.DifferentialDriveKinematics;
+import edu.wpi.first.math.kinematics.DifferentialDriveOdometry;
+import edu.wpi.first.math.kinematics.DifferentialDriveWheelSpeeds;
+import edu.wpi.first.units.BaseUnits;
+import edu.wpi.first.units.Unit;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Settings;
+
+/**<h1>i hate this **/
+public class Odometry extends SubsystemBase{
+
+
+    Pose2d pose = new Pose2d();
+
+    PigeonIMU gyro;
+    RelativeEncoder leftEncoder;
+    RelativeEncoder rightEncoder;
+    Drivetrain drivetrain = new Drivetrain();
```

Comment:
Pass into constructor instead of creating inside, so that the two classes are independent.